rotkehlchen/chain/ethereum/node_inquirer.py

```diff
@@ -281,17 +291,26 @@ def query_highest_block(self) -> BlockNumber:
 
         return BlockNumber(block_number)
 
-    def have_archive(self, requery: bool = False) -> bool:
-        if self.queried_archive_connection and requery is False:
-            return self.archive_connection
-
+    def _is_pruned(self, web3: Web3) -> bool:
+        try:
+            tx = web3.eth.get_transaction(HexStr(PRUNED_NODE_CHECK_TX_HASH.hex()))  # pylint: disable=no-member  # noqa: E501
```

Comment:
this is not needed. The typing in web3.eth.get_transaction is wrong. What you are doing here is you are turning a hexbytes, to hexstr and then pass it in, but hexbytes is accepted.

Just pass it as it and add a type ignore and an explanation as here: https://github.com/rotki/rotki/blob/c6bded6e4e4b44aea7771f6de8a90d8b87c508c7/rotkehlchen/chain/evm/node_inquirer.py#L690

Should work.